### ISSUE TITLE
Position of labels in notebooks 

### ISSUE THREAD

**Issue by kolibril13** [executablebooks/mystjs#268]
### Which area is this feature request for?

Website User Interface, Other

### Describe the feature you'd like to request

https://myst-tools.org/docs/mystjs/cross-references#targeting-cells says:
> You can label notebook cells using a comment at the top of the cell, using a `#| label:` syntax, or have this added directly in the notebook metadata for the cell.

## Suggestion:
I think it would be nice to have also the option to place the label not only at the first line, but also at the last line of a cell.

That would have two advantages:
1. For cells with long examples, label and image would stay close together and are easier to associate towards each other.
Currently, when there's a lot of other cell content in between label and image, the connection between these two might be more difficult to grasp with one glance.
Another benefit would be that one could search for `"#| label:` and would directly have the option to jump to all labelled images together with their tags. 

2. The first line of a jupyter cell is also the place of jupyter cell magic:
```python
%%time
import matplotlib.pyplot as plt
plt.plot([1,2], [10,20], c = "orange")
```
and cell magics can not be placed in a second line.
When there is a comment above the cell magic, then jupyter thinks it's not a cell magic but a linemagic (these are used in any lines of a notebook) and throws a `UsageError`.

<img width="312" alt="image" src="https://user-images.githubusercontent.com/44469195/221183318-e1b5baf6-0dad-45b5-ac42-93c9c327ca2a.png">

<img width="362" alt="image" src="https://user-images.githubusercontent.com/44469195/221183661-79cae5d5-2544-484f-a38b-ed0497d53472.png">

but 
```
%%time
import matplotlib.pyplot as plt
plt.plot([1,2], [10,20], c = "orange")
#| label: my-plot
```
will run without problems.

Having these two point in mind:
What do you think about having the label *ONLY* in the last line?

**Comment by kolibril13** [comment]
And another thought:
myst-nb uses cell tags to hide-input, hide-output and hide-cell.
See here -> https://myst-nb.readthedocs.io/en/latest/render/hiding.html
maybe these cell tags can be uses for mystjs for labels as well.

**Comment by rowanc1** [comment]
I will tackle this now -- very similar to #542!

**Comment by rowanc1** [comment]
I have done a partial improvement in #543. This allows the the `#| label:` comment to come after an ipython magic command, but it still requires it at the top of the cell.

Will leave this issue open and would love to hear from @fwkoch and others if we want that syntax to be supported anywhere.

**Comment by kolibril13** [comment]
that's great to hear, thanks for implementing this @rowanc1!

One more thought on label position:
Cells with plots can have many lines on code, and it can happen that cell label and cell output don't fit on the screen simultaneously:
<img width="896" alt="image" src="https://github.com/executablebooks/mystmd/assets/44469195/8ad59a8b-8ef1-4153-bc4c-5b8e4f650227">

I think in that particular case, a label at the button would make it easier to associate the connection between label and output.

**Comment by rowanc1** [comment]
Yep! I would love to hear from @fwkoch on if there are any downsides to opening this up to be allowed anywhere. :)

The potential downsides I see are (1) that it might be tempting to think that the labels are associated with the output, when they are associated with the cell. For example

```python
#| label: fig1
plot([1,2,3])
#| label: fig2
plot([5,6,7])
```

And (2): Should we support indented labels? e.g. inside of a function?

Curious on your thoughts on those potential issues?
